tomviz/ExternalPythonExecutor.cxx

```diff
@@ -174,4 +181,28 @@ QString ExternalPythonExecutor::commandLine(QProcess* process)
     .arg(process->arguments().join(" "));
 }
 
+void ExternalPythonExecutor::onStdOutReceived()
+{
+  QString s = m_process->readAllStandardOutput();
+  m_receivedStdOut += s;
+  // Since qDebug() will already print a newline, chop off the newline
+  // from the string if it is there.
```

Comment:
Won't this add a newline for those that don't have one ( for example write with `write` )?

Reply:
Yeah, it will. But getting `qDebug` to not print a newline seems to be tough - I haven't found a way to not do it. And we need `qDebug` or the output won't appear in the tomviz "Messages" box.

Reply:
Worse is it will remove a character that is not necessarily a `\n'?

Reply:
Yeah, but I'm not sure if Windows returns `\n`. It might return something else instead. I'll take a look.

Reply:
@cjh1 In the latest commit, I started using a [regular expression](https://github.com/OpenChemistry/tomviz/pull/2042/commits/2a095cc5af5828adeed93e3ba8b2d711cc6c2f75#diff-4976d2c27844f18f6ff4d5c847ee0d0fR193), which is probably a much better way to remove the newline at the end. The regular expression only matches with either a single `\n` (unix) or a single `\r\n` (windows) at the very end of the line. If there is one of these carriage returns at the end of the line, that carriage return gets removed.

A newline will still get added if a user prints via `sys.stdout.write()`, but that is kind of difficult to avoid here.

Just to record this here: if we ever decide in the future that we really don't want the newlines to be added, we can look into using the `vtk` printing functions. I think our tomviz message box is also hooked up to them.